packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -0,0 +1,234 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { FieldKey, EmptyKey, TreeValue } from "../../tree";
+import {
+	IEditableForest, TreeNavigationResult, mapCursorField, ITreeSubscriptionCursor, ITreeCursor,
+} from "../../forest";
+import { brand } from "../../util";
+import { TreeSchemaIdentifier } from "../../schema-stored";
+
+export const proxySymbol: unique symbol = Symbol("editable-tree proxy");
+
+/**
+ * A unique Symbol to get a method of {@link EditableTree} which returns a type of a node or its children.
+ *
+ * One can call `foo[getTypeSymbol]("bar")` to get a type of `bar` or
+ * without a field name i.e. `foo.bar[getTypeSymbol]()` for the same if `bar` is a non-primitive.
+ */
+export const getTypeSymbol: unique symbol = Symbol("editable-tree getType");
+
+/**
+ * This is a basis type for {@link EditableTree}.
+ *
+ * New features like e.g. to get an iterator over node's fields should be defined here
+ * and for each and every new method its own symbol should be created correspondingly.
+ * This prevents accidental mixin with user-defined string keys.
+ */
+export interface IEditableTreeSignature<T> {
```

Comment:
General naming note: it turns out this repo is inconsistent with its use of the "I" prefix on interfaces, and no one really sees it as offering value, so Tree might stop using that pattern. For now, you can do whichever you want and once a decision is reached, we can go update the names accordingly. Personally, I dislike the "I" prefix, and I think we might end up not using it, which is why I didn't use it in some of my suggested code.